tox.ini

```diff
@@ -210,34 +210,28 @@ commands =
 
 
 [testenv:lint]
-description =
-    lint the Python code with flake8 (code linter), black (code formatter), and
-    isort (sorting of imports)
+description = Run linters and type checks
 package = skip
 deps =
-    flake8
-    flake8-bugbear
-    black
+    ruff
     blackdoc
-    isort
 
 commands =
-    flake8 --extend-exclude "*/build/*" {[testenv]lint_folders}
-    black --check --diff {[testenv]lint_folders}
+    ruff format --diff {[testenv]lint_folders}
+    ruff check {[testenv]lint_folders}
     blackdoc --check --diff {[testenv]lint_folders}
-    isort --check-only --diff --extend-skip-glob "*/build/*" {[testenv]lint_folders}
+
 
 [testenv:format]
 description = Abuse tox to do actual formatting on all files.
 package = skip
 deps =
-    black
+    ruff
```

Comment:
should we pin ruff to a fixed version? How often do they update/change rules?

Reply:
IDK, if we start getting too much linter noise we can revisit this